### ISSUE TITLE
mpv freezes when seeking forward

### ISSUE THREAD

**Issue by youngunix** [mpv-player/mpv#11441]
### Important Information

Provide following Information:
- mpv version: 0.35.1-r1
- Linux Distribution and Version: Gentoo 17.1 
- Source of the mpv binary: emerge/portage
- If known which version of mpv introduced the problem: 0.35.1-r1
- Window Manager and version: xfwm 4.18.0
- GPU driver and version: nvidia-drivers 525.89.02
- Possible screenshot or video of visual glitches

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

Try to reproduce your issue with --no-config first. If it isn't reproducible
with --no-config try to first find out which option or script causes your issue.
^^^still having the issue even after trying the above suggestions.

Describe the reproduction steps as precise as possible. It's very likely that
the bug you experience wasn't reproduced by the developer because the workflow
differs from your own.
^^^I'm using yt-dlp 2023.03.04. To reproduce the issue; copy a youtube video's link, open terminal and type [mpv $youtube_link], once the video starts to play, then use the right arrow to seek forward (about 20sec) and the video freezes for a few minutes.  

### Expected behavior
mpv should not have an issue when trying to fast forward youtube videos. 

### Actual behavior
see above. 

### Log file

[mpv.log](https://github.com/mpv-player/mpv/files/10949989/mpv.log)

### Sample files
Use this link to play a video on mpv: https://www.youtube.com/watch?v=4wWI1iv4KEE

Sample files needed to reproduce this issue can be uploaded to https://0x0.st/
or similar sites. (Only needed if the issue cannot be reproduced without it.)
Do not use garbage like "cloud storage", especially not Google Drive.

**Comment by Dudemanguy** [comment]
Can''t reproduce. Is this still an issue with 0.36?

**Comment by llyyr** [comment]
This may be fixed by https://ffmpeg.org/pipermail/ffmpeg-devel/2023-August/313273.html if the issue only exists with hls streams

**Comment by rhendric** [comment]
I built ffmpeg with that patch and am still seeing the issue on mpv 0.36.

**Comment by youngunix** [comment]
Yes, the is still happening but not will all videos, but maybe that's due to either they are short or below 1080 resolution. 

Here is the output of `mpv -v` if it helps (I tried fixing the output, but github is not playing nice):

`[cplayer] Command line options: '-v'
[cplayer] mpv 0.35.1 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
[cplayer]  built on UNKNOWN
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.70.100
[cplayer]    libavcodec      58.134.100
[cplayer]    libavformat     58.76.100
[cplayer]    libswscale      5.9.100
[cplayer]    libavfilter     7.110.100
[cplayer]    libswresample   3.9.100
[cplayer] FFmpeg version: 4.4.4
[cplayer] 
[cplayer] Configuration: meson configure build -Dprefix=/usr -Dbuildtype=plain -Doptimization=plain
[cplayer] List of enabled features: alsa cdda cplayer cplugins dmabuf-interop-gl drm drm-is-kms dvdnav egl egl-drm egl-helpers egl-x11 ffmpeg gbm gl gl-x11 glibc-thread-name glob glob-posix gpl html-build iconv jpeg lcms2 libass libavdevice libdl libm libmpv-shared libplacebo libplacebo-next librt linux-fstatfs luajit manpage-build noexecstack plain-gl posix pulse shaderc stdatomic threads uchardet vdpau vdpau-gl-x11 vector vt.h vulkan x11 xv zlib
[cplayer] Built with NDEBUG.
[cplayer] Reading config file /etc/mpv/encoding-profiles.conf
[cplayer] Applying profile 'default'...
[cplayer] Reading config file /home/youngunix/.config/mpv/mpv.conf
[cplayer] Applying profile 'default'...
[cplayer] Setting option 'profile' = 'gpu-hq' (flags = 4)
[cplayer] Applying profile 'gpu-hq'...
[cplayer] Setting option 'scale' = 'spline36' (flags = 4)
[cplayer] Setting option 'cscale' = 'spline36' (flags = 4)
[cplayer] Setting option 'dscale' = 'mitchell' (flags = 4)
[cplayer] Setting option 'dither-depth' = 'auto' (flags = 4)
[cplayer] Setting option 'correct-downscaling' = 'yes' (flags = 4)
[cplayer] Setting option 'linear-downscaling' = 'yes' (flags = 4)
[cplayer] Setting option 'sigmoid-upscaling' = 'yes' (flags = 4)
[cplayer] Setting option 'deband' = 'yes' (flags = 4)
[cplayer] Setting option 'vo' = 'gpu' (flags = 4)
[cplayer] Setting option 'hwdec' = 'vdpau' (flags = 4)
[cplayer] Setting option 'deband' = 'no' (flags = 4)
[cplayer] Setting option 'cache' = 'yes' (flags = 4)
[cplayer] Setting option 'hls-bitrate' = 'max' (flags = 4)
[cplayer] Setting option 'demuxer-max-bytes' = '1000M' (flags = 4)
[cplayer] Setting option 'demuxer-max-back-bytes' = '1000M' (flags = 4)
[cplayer] Setting option 'script-opts' = 'try_ytdl_first=yes' (flags = 4)
[cplayer] Setting option 'script-opts' = 'ytdl_hook-ytdl_path=/usr/bin/yt-dlp' (flags = 4)
[cplayer] Setting option 'user-agent' = 'Mozilla/5.0 (X11; U; Unix; en-US) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.5672.93 Safari/537.36 Surf/0.6' (flags = 4)
[cplayer] Setting option 'ytdl-format' = '(best[height<=1080]/bestvideo+bestaudio)' (flags = 4)
[cplayer] Setting option 'save-position-on-quit' = '' (flags = 4)
[cplayer] Setting option 'v' = '' (flags = 8)
[cplayer] mpv 0.35.1 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
[cplayer]  built on UNKNOWN
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.70.100
[cplayer]    libavcodec      58.134.100
[cplayer]    libavformat     58.76.100
[cplayer]    libswscale      5.9.100
[cplayer]    libavfilter     7.110.100
[cplayer]    libswresample   3.9.100
[cplayer] FFmpeg version: 4.4.4
[cplayer] 
[cplayer] Usage:   mpv [options] [url|path/]filename
[cplayer] 
[cplayer] Basic options:
[cplayer]  --start=<time>    seek to given (percent, seconds, or hh:mm:ss) position
[cplayer]  --no-audio        do not play sound
[cplayer]  --no-video        do not play video
[cplayer]  --fs              fullscreen playback
[cplayer]  --sub-file=<file> specify subtitle file to use
[cplayer]  --playlist=<file> specify playlist file
[cplayer] 
[cplayer]  --list-options    list all mpv options
[cplayer]  --h=<string>      print options which contain the given string in their name
[cplayer] 
[cplayer] Set property: shared-script-properties -> 1
[cplayer] Set property: shared-script-properties -> 1
[cplayer] Set property: shared-script-properties -> 1
`

**Comment by sudoshindo** [comment]
Did you guys stop having this issue? Still happening on my end.
If I ever jump/seek forward the audio keeps playing but the video completely freezes.

**Comment by youngunix** [comment]
> Did you guys stop having this issue? Still happening on my end.
> If I ever jump/seek forward the audio keeps playing but the video completely freezes.

Yes, it's still happening on my system. I suspect it's a connection latency caused by ffmpeg.

**Comment by Mezmathics** [comment]
> > Did you guys stop having this issue? Still happening on my end.
> > If I ever jump/seek forward the audio keeps playing but the video completely freezes.
> 
> Yes, it's still happening on my system. I suspect it's a connection latency caused by ffmpeg.

After reading this thread and also noticing for myself that this is an issue with some videos than others, I noticed it actually depends not so much on the videos but the format yt-dlp picks up. even if you have specific configuration there are many options that satisfy that configuration that yt-dlp can pick up. For example if you specify 

`--ytdl-format="bestvideo[ext=mp4][height<=?1080]+bestaudio[ext=m4a]" there`

there are several formats that yt-dlp can pick from that satisfy this and some of them have this issues while others don't. For each video, you can check the full list of available formats with 

`yt-dlp -F [YOUTUBE LINK]`

each format has an associated id and it shows all other information such as extension and if it's video or audio

With YouTube currently, every video that I tried that had this issue was resolved by first making sure  I set `--script-opts-append=ytdl_hook-try_ytdl_first=yes` I picked id 137 for the video which is one of the mp4 1080p formats if it's available and id 140 for audio which seems to be m4a. on all the videos I checked that have a minimum 1080p resolution this ids were available so I added in my mpv.conf file 

`--script-opts-append=ytdl_hook-try_ytdl_first=yes

[youtube]
profile-cond=path:find('youtu%.?be')
--ytdl-format="137+140" `

which sets `--ytdl-format="137+140"` only for videos containing "youtube" or "youtu.be" in their path and the formats specified are available. If not it gets ignored. So far this has worked for me

**Comment by sudoshindo** [comment]
@Mezmathics can you share what your current `mpv.conf` looks like?

**Comment by youngunix** [comment]
I set the height to 720 and it seems to remediate the issue. As I suspected, the buffer either has a limit when steaming videos >=1080 or  Youtube (CDN) are throttling their upload speed. The 720 quality is not bad, but for some, it might be a bummer. Unfortunately, we have to live with it for now until someone finds a fix.

**Comment by sudoshindo** [comment]
@youngunix share your conf?

**Comment by youngunix** [comment]
> @youngunix share your conf?

Are you having issues too or you just need a sample 'mpv.conf'?

**Comment by sudoshindo** [comment]
I have the same issue, otherwise I wouldn't be in this thread.

**Comment by youngunix** [comment]
> I have the same issue, otherwise I wouldn't be in this thread.

Next time lead with that, don't just drop in and demand info. This is not the proper way of engaging in technical discussions (or any other ones for that matter).

**Comment by sudoshindo** [comment]
i don't see you in the top contributor graph for this repo

**Comment by youngunix** [comment]
> i don't see you in the top contributor graph for this repo

You insinuate that you are a "contributor," yet you contributed nothing to this discussion. Instead, you polluted it with your toxic personality. 

Little kid, go watch some cartoons.

**Comment by zunjack** [comment]
Is this unfixable?
Please only devs reply, thanks.

**Comment by ydrojd** [comment]
Setting the video container to webm instead of mp4 fixed this for me. 

--ytdl-format="bestvideo[ext=webm]+bestaudio[ext=m4a]"

**Comment by sudoshindo** [comment]
@ydrojd thank you very very very very very much for giving an actual solution instead of floundering around and arguing with people

**Comment by sudoshindo** [comment]
@ydrojd thank you very very very very much for giving an actual solution instead of being a dickhead and arguing with people

**Comment by youngunix** [comment]
> [@ydrojd](https://github.com/ydrojd) thank you very very very very much for giving an actual solution instead of being a dickhead and arguing with people

better than a pussy ass coward like you, you fucking faggot.

**Comment by Akemi** [comment]
this is quite enough. cool your heads down.
